enterprise/tpa_pipeline.py

```diff
@@ -0,0 +1,164 @@
+"""
+Module provides elements to be used in third-party auth pipeline.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.core.urlresolvers import reverse
+from django.shortcuts import redirect
+from django.utils.translation import ugettext as _
+
+from enterprise.models import EnterpriseCustomer, EnterpriseCustomerUser, UserDataSharingConsentAudit
+from enterprise.utils import NotConnectedToEdX
+
+try:
+    from social.pipeline.partial import partial
+except ImportError:
+    from enterprise.utils import null_decorator as partial
+
+try:
+    from third_party_auth.provider import Registry
+except ImportError:
+    Registry = None
+
+
+def get_enterprise_customer_for_request(request):
+    """
+    Get the EnterpriseCustomer associated with a particular request.
+    """
+    pipeline = request.session.get('partial_pipeline')
+    return get_ec_for_running_pipeline(pipeline)
```

Comment:
If `pipeline` is `None` (as in a blank session), this is causing a 500 error for me:

```python
  File "/edx/app/edxapp/edx-platform/common/djangoapps/util/enterprise_helpers.py", line 63, in insert_enterprise_fields
    add_data_sharing_consent_field(request, form_desc)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/util/enterprise_helpers.py", line 71, in add_data_sharing_consent_field
    enterprise_customer = get_enterprise_customer_for_request(request)
  File "/edx/src/edx-enterprise/enterprise/tpa_pipeline.py", line 29, in get_enterprise_customer_for_request
    return get_ec_for_running_pipeline(pipeline)
  File "/edx/src/edx-enterprise/enterprise/tpa_pipeline.py", line 41, in get_ec_for_running_pipeline
    provider = Registry.get_from_pipeline(pipeline)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/third_party_auth/provider.py", line 72, in get_from_pipeline
    if enabled.is_active_for_pipeline(running_pipeline):
  File "/edx/app/edxapp/edx-platform/common/djangoapps/third_party_auth/models.py", line 177, in is_active_for_pipeline
    return self.backend_name == pipeline['backend']
TypeError: 'NoneType' object has no attribute '__getitem__'
```

To reproduce: Open http://localhost:8000/login in an incognito window. Boom, 500 error.